config/logging.go

```diff
@@ -0,0 +1,203 @@
+package config
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+
+	"github.com/pelletier/go-toml/v2"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
+	"github.com/smartcontractkit/chainlink-testing-framework/utils/net"
+)
+
+//go:embed tomls/logging_default.toml
+var DefaultLoggingConfig []byte
+
+type LoggingConfig struct {
+	TestLogCollect *bool            `toml:"test_log_collect"`
+	RunId          *string          `toml:"run_id"`
+	Loki           *LokiConfig      `toml:"Loki"`
+	Grafana        *GrafanaConfig   `toml:"Grafana"`
+	LogStream      *LogStreamConfig `toml:"LogStream"`
+}
+
+func (l *LoggingConfig) Validate() error {
+	if l.LogStream != nil {
+		if err := l.LogStream.Validate(); err != nil {
+			return fmt.Errorf("invalid log stream config: %w", err)
+		}
+	}
+
+	if l.Grafana != nil {
+		if err := l.Grafana.Validate(); err != nil {
+			return fmt.Errorf("invalid grafana config: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (l *LoggingConfig) ApplyOverrides(from *LoggingConfig) error {
```

Comment:
We can do simple multi-marshal to the same pointer, example
```
package main

import (
	"github.com/davecgh/go-spew/spew"
	"github.com/pelletier/go-toml/v2"
	"os"
)

type Config struct {
	Group1 *Group1 `toml:"group_1"`
}

type Group1 struct {
	Name         string  `toml:"name"`
	Surname      string  `toml:"surname"`
	UniqueWeapon string  `toml:"unique_weapon"`
	Inventory    []int64 `toml:"inventory"`
}

func readAndOverride(prevCfg *Config, name string) (*Config, error) {
	f, err := os.ReadFile(name)
	if err != nil {
		panic(err)
	}
	if err := toml.Unmarshal(f, &prevCfg); err != nil {
		panic(err)
	}
	spew.Dump(prevCfg)
	return prevCfg, nil
}

func main() {
	cfg := &Config{}
	cfg1, err := readAndOverride(cfg, "config_base.toml")
	if err != nil {
		panic(err)
	}
	cfg2, err := readAndOverride(cfg1, "config_product.toml")
	if err != nil {
		panic(err)
	}
	_, err = readAndOverride(cfg2, "config_override.toml")
	if err != nil {
		panic(err)
	}
}
```
Output
```
(*main.Config)(0x1400005a038)({
 Group1: (*main.Group1)(0x1400007a140)({
  Name: (string) (len=4) "John",
  Surname: (string) (len=4) "Snow",
  UniqueWeapon: (string) (len=5) "sword",
  Inventory: ([]int64) (len=2 cap=16) {
   (int64) 1,
   (int64) 2
  }
 })
})
(*main.Config)(0x1400005a038)({
 Group1: (*main.Group1)(0x1400007a140)({
  Name: (string) (len=4) "John",
  Surname: (string) (len=4) "Snow",
  UniqueWeapon: (string) (len=5) "sword",
  Inventory: ([]int64) (len=1 cap=16) {
   (int64) 3
  }
 })
})
(*main.Config)(0x1400005a038)({
 Group1: (*main.Group1)(0x1400007a140)({
  Name: (string) (len=4) "John",
  Surname: (string) (len=9) "Malkovich",
  UniqueWeapon: (string) (len=5) "sword",
  Inventory: ([]int64) (len=1 cap=16) {
   (int64) 999
  }
 })
})
```
That can reduce amount of logic in cfg code we need to write and maintain, and overriding this way is more straightforward

Reply:
This applies to all config structures